site/source/tutorials/reactive_programming.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: false
+title: Reactive programming
+overview: Dojo 2 embraces reactive programming patterns like top-down property propagation. This article gives a high level overview of the reactive patterns in Dojo 2 and how they are different from more traditional patterns.
+---
+# Reactive programming
+
+Reactive programming is concernedwith handling change - specifically, the _propagation of change_. As web applications have become more and more asynchronous, keeping track of application state has become increasingly complex. With traditional programming patterns, we rely on event listeners or try to keep track of application state manually.
+
+Consider a banking application where the user's account balance might appear in several different places on the screen. After the user makes a withdrawal, we want the account balance to update.
+
+## Imperative approach
+
+Traditionally, we might use an event emitter. Each time the balance is updated we emit an event, and any interested parties can listen for these events.
+
+![Banking](resources/imperative.png)
+
+```javascript
+// assume some sort of application wide messageBus
+
+function makeWithdrawal() {
+    // make the withdrawal
+    messageBus.emit('account-balance-updated', newBalance);
+}
+
+// at different points in the UI
+messageBus.on('account-balance-updated', (newBalance) => {
+    accountDomElement.innerText = newBalance;
+});
+```
+
+With an event emitter, any interested widget can listen for updates to the account balance.  This pattern has served well in the past, but there are a few problems with this approach:
+
+*   We need to remember to emit `account-balance-updated` every time the account balance is changed.
+*   We need to remember to listen for balance updates.
+*   We need to have a facility to **remove** the event listeners when they are no longer needed.
+
+## Reactive approach
+
+Dojo 2 is built around unidirectional, top-down property propagation It is the parent widget's job to pass properties down to its children. In fact, a child widget has no direct reference to a parent widget! When a property changes, widgets are re-rendered (using an efficient virtual DOM) to reflect the updated state. 
```

Comment:
It -> it

Reply:
propagation It is -> propagation where It is